locust/core.py

```diff
@@ -556,19 +560,59 @@ def _set_setup_flag(cls):
     def _set_teardown_flag(cls):
         cls._teardown_is_set = True
     
-    def run(self, runner=None):
+    def run(self):
+        self._state = LOCUST_STATE_RUNNING
         task_set_instance = self._task_set(self)
         try:
+            if hasattr(self, "on_start"):
```

Comment:
would it be possible to define an on_start/on_stop method on the TaskSet base class that does nothing?

Reply:
Yeah, that might be better. I guess that any performance difference should be negligible (haven't measure so I don't even know which one is fastest).